yt_dlp/extractor/youtube.py

```diff
@@ -1900,50 +2142,107 @@ def get_text(x):
+            ytm_query.update(self._generate_player_context(sts))
+
+            ytm_player_response = self._extract_response(
+                item_id=video_id, ep='player', query=ytm_query,
+                ytcfg=ytm_cfg, headers=ytm_headers, fatal=False,
+                default_client=ytm_client,
+                note='Downloading web remix player API JSON')
+
+            ytm_streaming_data = try_get(ytm_player_response, lambda x: x['streamingData']) or {}
 
-        player_response = None
         if webpage:
             player_response = self._extract_yt_initial_variable(
                 webpage, self._YT_INITIAL_PLAYER_RESPONSE_RE,
                 video_id, 'initial player response')
 
-        ytcfg = self._extract_ytcfg(video_id, webpage)
+        # Backup request for player_response if regex fails
+        # and/or Youtube removes player response on initial page
         if not player_response:
-            player_response = self._call_api(
-                'player', {'videoId': video_id}, video_id, api_key=self._extract_api_key(ytcfg))
+            sts = self._extract_signature_timestamp(ytcfg)
+            yt_client = 'WEB'
+            ytpcfg = ytcfg
+            ytp_headers = headers
+            if not sts:
+                # Android client already has signature descrambled
+                # See: https://github.com/TeamNewPipe/NewPipeExtractor/issues/562
+                self.report_warning('Falling back to mobile client for player API.')
+                ytpcfg = self._get_default_ytcfg('ANDROID')
+                yt_client = 'ANDROID'
+                ytp_headers = self._generate_api_headers(ytpcfg, identity_token, syncid, yt_client)
+
+            yt_query = {'videoId': video_id}
+            yt_query.update(self._generate_player_context(sts))
+            player_response = self._extract_response(
+                item_id=video_id, ep='player', query=yt_query,
+                ytcfg=ytpcfg, headers=ytp_headers, fatal=False,
+                default_client=yt_client,
+                note='Downloading player API JSON'
+            )
 
+        # Age-gate workarounds
         playability_status = player_response.get('playabilityStatus') or {}
         if playability_status.get('reason') == 'Sign in to confirm your age':
             pr = self._parse_json(try_get(compat_parse_qs(
                 self._download_webpage(
                     base_url + 'get_video_info', video_id,
-                    'Refetching age-gated info webpage',
-                    'unable to download video info webpage', query={
-                        'video_id': video_id,
-                        'eurl': 'https://youtube.googleapis.com/v/' + video_id,
-                        'html5': '1',
-                    }, fatal=False)),
+                    'Refetching age-gated info webpage', 'unable to download video info webpage',
+                    query=self._get_video_info_params(video_id), fatal=False)),
```

Comment:
Keeping the old bypass in for the time being, but have the embed-only one present for a fallback